components/TransLine.tsx

```diff
@@ -15,7 +15,11 @@ export interface TransLineProps extends TransLineContent {
 function resolveUrl(link: I18nString, userArrivedFromUioMobile: boolean) {
   // Special case for UIO homepage links.
   if (link === 'uio-home') {
-    const uioHomeLink = userArrivedFromUioMobile ? getUrl('uio-home-url-mobile') : getUrl('uio-home-url-desktop')
+    // Optional environment-specific links back to the UIO landing page, used by EDD testing
+    const uioLandingUrl = process.env.URL_UIO_LANDING ?? getUrl('uio-home-url-desktop')
+    const uioMobileLandingUrl = process.env.URL_UIOMOBILE_LANDING ?? getUrl('uio-home-url-mobile')
+
+    const uioHomeLink = userArrivedFromUioMobile ? uioMobileLandingUrl : uioLandingUrl
     if (uioHomeLink) {
```

Comment:
@rocketnova I don't understand this if statement-- is it to protect against the possibility of uio-home-url-desktop/mobile, not existing as a string? If they do exist, then uioHomeLink will always be true right?

Reply:
Typescript will yell if we don't narrow here, because the code doesn't know those json strings _definitely_ exist - we have a ticket to see if there's some way to assert the json strings do exist #338

Reply:
`getUrl()` can return undefined. I'm not positive what happens if `public/urls.json` disappears or is malformed or has a read-error, but there's definitely the possibility that `public/urls.json` exists and the `linkKey` passed in doesn't exist in the file.

Reply:
> Typescript will yell if we don't narrow here, because the code doesn't know those json strings _definitely_ exist - we have a ticket to see if there's some way to assert the json strings do exist #338

Agreed that we should know at build time, but I think the check is important in case something horrible and unexpected happens at runtime

Reply:
Gotcha, this all makes sense. Thanks!